presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java

```diff
@@ -48,4 +55,39 @@ public void testIsViewFileSystem()
         assertTrue(isViewFileSystem(CONTEXT, hdfsEnvironment, viewfsPath));
         assertFalse(isViewFileSystem(CONTEXT, hdfsEnvironment, nonViewfsPath));
     }
+
+    @Test void testCreateTemporaryPathOnViewFS()
+    {
+        HdfsEnvironment hdfsEnvironment = createTestHdfsEnvironment(new HiveClientConfig(), new MetastoreClientConfig());
+        Path viewfsPath = new Path("viewfs://ns-default/test-dir");
+
+        // ViewFS check requires the mount point config
+        hdfsEnvironment.getConfiguration(CONTEXT, viewfsPath).set("fs.viewfs.mounttable.ns-default.link./test-dir", "file://" + Files.createTempDir());
+
+        Path temporaryPath = createTemporaryPath(SESSION, CONTEXT, hdfsEnvironment, viewfsPath);
+
+        assertEquals(temporaryPath.getParent().toString(), "viewfs://ns-default/test-dir/.hive-staging");
```

Comment:
could we have testcases for both targetPath exist and not exist?

Reply:
test cases added